### PR TITLE
Cloud Monitoring: Update GroupBy fields to use experimental UI components

### PR DESCRIPTION
**Pull request by kevinwcyu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Relates to #44431

**Special notes for your reviewer**:
The base branch is currently set to `kevinwcyu/44431-cloudmonitoring-alignment-experimental-ui`. After that one is merged, I'll update the base branch to `main`

Current group by fields
Current Cloud Monitoring group by fields

Updated group by fields (the width of the `Group by` selector is now set to match the other selector widths. Previously it was set to be wider so it would be aligned with the two input fields in the row below it.
Updated Cloud Monitoring group by fields

---

The Experimental folder contains all of the components as they are being worked on. Currently the components within the directory are all just copied from the directory one level above. In general the only changes involve adding `<EditorRows>`, `<EditorRow>` (not plural), and `<EditorField>` components.

Eventually, we should be able to just move the components in the Experimental directory up one level and replace the existing files and remove the `cloudMonitoringExperimentalUI` feature toggle.

### CODE REVIEW

**Review by andresmgot — COMMENTED** [review]
I noticed a small bug here. If the name of the label is bigger than the size of the select box, the `x` icon is not accessible, making it impossible to remove it:

Screenshot from 2022-06-10 12-04-57

Can we make it grow horizontally?

**Comment by kevinwcyu** [comment]
> I noticed a small bug here. If the name of the label is bigger than the size of the select box, the `x` icon is not accessible, making it impossible to remove it:
> 
> Screenshot from 2022-06-10 12-04-57
> 
> Can we make it grow horizontally?

@andresmgot I've pushed a change to make the width `auto`. It grows horizontally and will wrap the `Group by function` field to the next row. See the screenshots below.

Screen Shot 2022-06-10 at 10 33 00 AM

Screen Shot 2022-06-10 at 10 33 10 AM

Let me know what you think about this. Another potential option is setting a `maxWidth` on the MultiSelect component, but we'd need a custom style for that.

**Review by sunker — COMMENTED** [review]
I'd also set the width for the "Group by function" input to _auto_. 

The existing UI uses a fixed width for labels, inputs and selects in an attempt to align things horizontally so that it would look a little less chaotic. With the experimental UI, there should be no need for that. Think you can remove the fixed width from all selects.

**Comment by andresmgot** [comment]
> With the experimental UI, there should be no need for that. Think you can remove the fixed width from all selects.

That's a good point, I was going to suggest to add a min-width to match the rest but it's probably better to just remove the width for all. Since the fields would take less space, we can potentially merge the "group by" and "alignment" rows in one.

**Comment by sunker** [comment]
I think the reason the editor looks the way it does with preprocessing, group by and alignment on separate rows is that Google wanted us to build something that look similar to their dashboard edit mode (this is not the same as Metrics explorer that we use in deep links). 

Screenshot from 2022-06-13 10-24-47

Rows in the experimental UI takes up a bit more space vertically, so it might be good to combine one or two rows though. If so, we could consider putting the pro-processing field in the alignment row since they're all about alignment of data points. What do you all say we revisit this once all fields have been migrated to experimental UI and we have get to see everything together?

**Comment by andresmgot** [comment]
> What do you all say we revisit this once all fields have been migrated to experimental UI and we have get to see everything together?

makes sense, thanks for the background!

**Comment by kevinwcyu** [comment]
> What do you all say we revisit this once all fields have been migrated to experimental UI and we have get to see everything together?

Sounds good.

File: public/app/plugins/datasource/cloud-monitoring/components/Experimental/GroupBy.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+export interface Props {
+  refId: string;
+  variableOptionGroup: SelectableValue<string>;
+  labels: string[];
+  metricDescriptor?: MetricDescriptor;
+  onChange: (query: MetricQuery) => void;
+  query: MetricQuery;
+}
+
+export const GroupBy: FunctionComponent<Props> = ({
```
Comment (andresmgot) [inline public/app/plugins/datasource/cloud-monitoring/components/Experimental/GroupBy.tsx:22]: are there tests for this?

**Review by andresmgot — APPROVED** [review]

File: public/app/plugins/datasource/cloud-monitoring/components/Experimental/GroupBy.tsx [hunk 2]
```diff
@@ -38,7 +38,6 @@ export const GroupBy: FunctionComponent<Props> = ({
         <EditorField
           label="Group by"
           tooltip="You can reduce the amount of data returned for a metric by combining different time series. To combine multiple time series, you can specify a grouping and a function. Grouping is done on the basis of labels. The grouping function is used to combine the time series in the group into a single time series."
-          htmlFor={`${refId}-group-by`}
```
Comment (andresmgot) [inline public/app/plugins/datasource/cloud-monitoring/components/Experimental/GroupBy.tsx]: why did you remove this `htmlFor`?

**Review by andresmgot — APPROVED** [review]
LGTM

File: public/app/plugins/datasource/cloud-monitoring/components/Experimental/GroupBy.tsx (hunk 2 above)
Comment (kevinwcyu) [reply public/app/plugins/datasource/cloud-monitoring/components/Experimental/GroupBy.tsx]: The `EditorField` automatically gets the child input id. The `getByLabelText` in the tests would fail if there was no input element associated with the label.
